### DISCUSSION TITLE
Is there a reason `verifyMessage` has two different signatures/implementations?

### DISCUSSION THREAD

**Discussion by L-Kov** [wevm/viem#1125] [Question] [answered]
PR #610 added support for signing, verifying and recovering hash data representation of messages. It updated the related functions to accept a `SignableMessage` type versus just a string. The exported `verifyMessage` function still supports `string` message types versus the extended `SignatbleMessage` type. Is this by design? 

See public actions implementation [here](https://github.com/wagmi-dev/viem/blob/7da5224467ca5a6368f27734a01b56f53c2c0106/src/actions/public/verifyMessage.ts#L32) (supports `SignableMessage` message type)

See signature util implemenation [here](https://github.com/wagmi-dev/viem/blob/7da5224467ca5a6368f27734a01b56f53c2c0106/src/utils/signature/verifyMessage.ts#L32) (only allows `string` message type)

**Comment by jxom** [comment, marked as answer]
Ah yes. We need to update this in the signature util.
